### ISSUE TITLE
WebView refreshes to the main page when rotating device

### ISSUE THREAD

**Issue by mmc20520** [openhab/openhab-android#2845]
### Actual behaviour
WebView refreshes to the main page when rotating device

For example, randomly browser in the webview. Before rotation:
<img width="292" alt="Screen Shot 2022-03-04 at 4 23 15 PM" src="https://user-images.githubusercontent.com/100833575/156726761-ac2bac59-3b35-46e9-bda8-1907d626df52.png">

After rotation:

<img width="623" alt="Screen Shot 2022-03-04 at 4 23 27 PM" src="https://user-images.githubusercontent.com/100833575/156726786-d7185dca-e2d3-4180-8deb-6490ad82edcc.png">

### Expected behaviour
The loaded web page should remain unchanged

 
### Steps to reproduce
Already mentioned above

### Can you reproduce the issue in demo mode?
yes

#### Client
* Android version: 10
* Device model: emulator
* App version : 2.19.0 
* Build flavor : 
* Device language: 

#### Server
* Server version: 
* Reverse Proxy: 
* Authentication method : 

#### openHAB Cloud
* Self hosted cloud or myopenhab.org: 
* openHAB Cloud version : 
* Authentication method : 

#### App log
<details>
  <summary>Click to expand</summary>

```
Please add the app log if the issue is not a pure UI issue or it cannot be reproduced in demo mode.
Open the app, go to `Settings` => `View log` and insert the log here.
```

</details>

#### openHAB Server log
<details>
  <summary>Click to expand</summary>

```
Not required.
If you have installed openHAB via `apt-get`, the log is located under `/var/log/openhab2/openhab.log`
```

</details>

**Comment by mueller-ma** [comment]
Duplicate of https://github.com/openhab/openhab-android/issues/2606

**Comment by littlecd** [comment]
Is it fixed now? Seems I can still reproduce the issue in 3.7.0. I can also observe the issue from Frontail.

**Comment by vguna** [comment]
Also still happening for me on OH 4.1.0 and app 3.10.

![GIF_20240310_183251_807](https://github.com/openhab/openhab-android/assets/3257311/3ac4ebdb-0af7-4f7a-aead-ccd3deb17af4)

**Comment by maniac103** [comment]
@florian-h05 ping as desired ;-)

**Comment by florian-h05** [comment]
Thanks for pinging.

First question: Is it really required and cannot be avoided that the Android WebView reloads on every orientation change?

Second: Main UI still is not perfect with encoding its current navigation into the navigation route, it now does this for tabs as well (not sure if I added that before or after 4.2.0), but for the popup based stuff like the semantic model cards, this is not implemented.
Overall it’s not easy to do that I think due to the way framework7 opens its modals. Those models usually modify the navigation route, but there is no actual route defined for them in the route so you cannot load the application into that modal.
I can have a look at improving that if you say there is no other way though.

**Comment by maniac103** [comment]
> First question: Is it really required and cannot be avoided that the Android WebView reloads on every orientation change?

Yes. This is related to having differing layouts for portrait and landscape orientation. The activity is recreated and all views within are recreated with it.
One can avoid recreation by specifying the 'orientation' value for [configChanges in the manifest](https://developer.android.com/guide/topics/manifest/activity-element#config), but that only works properly for pure WebView activities (that is, the only layout element is a WebView), which isn't the case for us, since for sitemaps we switch between a one and a two pane layout.
